### DISCUSSION TITLE
Does react navigation works with New architecture enabled?

### DISCUSSION THREAD

**Discussion by BraveEvidence** [react-navigation/react-navigation#10670] [Q&A] [answered]
Does react navigation works with fabric and turbo modules?

**Comment by kacperkapusciak** [comment, marked as answer]
Hi @PritishSawant :wave:

`react-navigation` as its own is fully written in JavaScript/TypeScript and doesn't have any native code but it uses native dependencies as `react-native-gesture-handler`, `react-native-screens`, `react-native-safe-area-context` and `react-native-reanimated`. It's up to these packages to support the new architecture.  

As of today, you *can* run `react-navigation` on the new architecture but I wouldn't expect full feature parity or production-level stability.

Cheers

**Reply by frankcalise** [reply]
I believe this is still an issue, though, as setNativeProps isn't supported - https://github.com/react-navigation/react-navigation/issues/10720

**Comment by BraveEvidence** [comment]
@kacperkapusciak Thanks, I think all the libraries which you mentioned are fabric compatible.

**Reply by frankcalise** [reply]
@PritishSawant reanimated does not yet, v3 will support this but it is still a release candidate
